spec.bs

```diff
@@ -1356,6 +1467,204 @@ To <dfn>build an interest group passed to generateBid</dfn> given an [=interest
+    1. Let |decodedSignedAdditionalBid| be the result of running [=forgiving-base64 decode=] with
+      |encodedAdditionalBid|.
+    1. If |decodedSignedAdditionalBid| is failure, [=iteration/continue=].
+    1. Increment |auctionConfig|’s [=auction config/scoring dependency count=].
+    1. Let |parsedSignedAdditionalBid| be the result of [=parse a JSON string to an infra value=]
+      with |decodedSignedAdditionalBid|.
+    1. Decrement |auctionConfig|’s [=auction config/scoring dependency count=], and
+      [=iteration/continue=] if any of the following conditions hold:
+      * |parsedSignedAdditionalBid| is not a [=map=];
+      * |parsedSignedAdditionalBid|["bid"] does not [=map/exist=], or it's not a [=string=];
+      * |parsedSignedAdditionalBid|["signatures"] does not [=map/exist=], or it's not a [=list=].
+    1. Let |signatures| be a new [=list=] of [=signed additional bid signatures=].
+    1. Let |decodeSignatureFailed| be false.
+    1. [=list/For each=] |sig| of |parsedSignedAdditionalBid|["signatures"]:
+      1. Set |decodeSignatureFailed| to true and [=iteration/break=] if any of the following
+        conditions hold:
+        * |sig| is not a [=map=];
+        * |sig|["key"] does not [=map/exist=], or it's not a [=string=];
+        * |sig|["signature"] does not [=map/exist=], or it's not a [=string=].
+      1. Let |maybeKey| be the result of running [=forgiving-base64 decode=] with |sig|["key"].
+      1. Let |maybesignature| be the result of running [=forgiving-base64 decode=] with
+        |sig|["signature"].
+      1. Set |decodeSignatureFailed| to true and [=iteration/break=] if any of the following
+        conditions hold:
+        * |maybeKey| is failure, or its [=byte sequence/length=] is not 32;
+        * |maybesignature| is failure, or its [=byte sequence/length=] is not 64;
+      1. Let |signature| be a [=signed additional bid signatures=], whose
+        [=signed additional bid signature/key=] is |maybeKey|, and
+        [=signed additional bid signature/signature=] is |maybesignature|.
+      1. [=list/Append=] |signature| to |signatures|.
+    1. If |decodeSignatureFailed| is true, then decrement |auctionConfig|’s
+      [=auction config/scoring dependency count=], and [=iteration/continue=].
+    1. Let |decodedAdditionalBid| be the result of [=decode an additional bid=] with
+      |parsedSignedAdditionalBid|["bid"], |auctionConfig| and |topLevelAuctionConfig|.
+    1. Decrement |auctionConfig|’s [=auction config/scoring dependency count=], and
+      [=iteration/continue=] if any of the following conditions hold:
+      * |decodedAdditionalBid| is failure;
+      * |decodedAdditionalBid|'s [=additional bid/bid=]'s [=generated bid/bid=]'s
+        [=bid with currency/currency=] is not the result of running [=look up per-buyer currency=]
+        with |auctionConfig|.
+    1. Let |validSignatureIndices| be a new [=list=] of integers.

```

Comment:
I think our use of indices to point to the valid signatures is an optimization and an implementation detail.

Would it reasonable to say, here, 
`[=ordered set/append=] |signatures|[|i|]'s [=signed additional bid signature/key=] to |verifiedKeys|`
(`Let |verifiedKeys| be an [=ordered set=] of [=string=]`)

And then later, instead of lines 1533-1537, (Let |KeyMatchedValidSignature| be false.  through If |KeyMatchedValidSignature| is false), replace that with something like,
`If |verifiedKeys| [=ordered set/contains=] |additionalBidKey| is false, then [=iteration/continue=]`?

Reply:
`verifiedKeys`, do you mean `verifiedSignatures`? Since it's checking signatures instead of keys here.
Done.